### DISCUSSION TITLE
PQ AKE performance

### DISCUSSION THREAD

**Discussion by keskinadem** [strongswan/strongswan#2476] [Q&A] [answered]
I'm just asking because I'm curious.

Before PQ, KE in IKE_SA_INIT can be completed in single exchange but with PQ AKE, IKE_INTERMEDIATE rises and exchange count doubles  (for example frodo). but PQ AKE's are larger so it should be fragmented. Lets suppose PQ AKE completed in in 30 fragmented messages(15 for initiator, 15 for responder). The elapsed time between acquire received and child_sa established is not increases accordingly. The observed network packages in wireshark increases maybe 10 times but duration increases just two times. What makes this PQ AKE so fast?

**Comment by tobiasbrunner** [comment, marked as answer]
Those fragments are not individually acked/exchanged and they don't require any additional expensive computing. So if no retransmissions are necessary, it's basically just like one additional IKE exchange even though 30 messages might get sent/received between the peers.
